census_view.cpp

```diff
@@ -1325,6 +2345,64 @@ void CensusDVCView::update_visible_columns()
         }
 }
 
+void CensusGridView::update_visible_columns()
+{
+    // Display exactly those columns whose rows aren't all identical. For
+    // this purpose, consider as "rows" the individual cells--and also the
+    // case and class defaults, even though they aren't displayed in rows.
+    // Reason: although the case and class defaults are hidden, they're
+    // still information--so if the user made them different from any cell
+    // wrt some column, we respect that conscious decision.
+    std::vector<std::string> const& all_headers(case_parms()[0].member_names());
+    std::vector<int> new_visible_columns;
+    int column = 0;
+    for(auto const& header : all_headers)
+        {
+        if(column_value_varies_across_cells(header))
+            {
+            new_visible_columns.push_back(column);
+            }
+        ++column;
+        }
+
+    if(new_visible_columns != grid_table_->get_visible_columns())
+        {
+        auto const cursor_row = grid_window_->GetGridCursorRow();
+        auto const cursor_col = grid_window_->GetGridCursorCol();
+
+        auto const columns_count     =
+            lmi::ssize(grid_table_->get_visible_columns()) + 1;
+        auto const new_columns_count =
+            lmi::ssize(new_visible_columns)                + 1;
+
+        wxGridUpdateLocker grid_update_locker(grid_window_);
+
+        if(grid_window_->IsCellEditControlEnabled())
```

Comment:
AFAICS this is unnecessary and we can always just call `DisableCellEditControl()` directly, it won't do anything if there is no edit control currently shown.